terraform/gcp/main.tf

```diff
@@ -0,0 +1,202 @@
+
+      type = (count.index >= var.BOR_VALIDATOR_COUNT + var.BOR_SENTRY_COUNT) ? var.BOR_ARCHIVE_VOLUME_TYPE : var.BOR_VOLUME_TYPE
+    }
+  }
+
+  metadata = {
+    ssh-keys = "${var.USER}:${file(var.GCE_PUB_KEY_FILE)}"
+  }
+
+  network_interface {
+    network    = google_compute_network.vpc_network.name
+    subnetwork = google_compute_subnetwork.public-subnetwork.name
+
+    access_config {
+      //IP
+      nat_ip = google_compute_address.bor_static_ip[count.index].address
+    }
+  }
+  tags = ["matic-cli"]
+  labels = {
+    name     = "polygon-matic"
+    instance = "bor"
+  }
+}
+
+resource "google_compute_instance" "erigon_node_server" {
+
+  count = (var.DOCKERIZED == "yes") ? 0 : (var.ERIGON_VALIDATOR_COUNT + var.ERIGON_SENTRY_COUNT + var.ERIGON_ARCHIVE_COUNT)
+
+  name = "${var.VM_NAME}-erigon-${count.index + 1}"
+
+  machine_type = (count.index >= var.ERIGON_VALIDATOR_COUNT + var.ERIGON_SENTRY_COUNT) ? var.ERIGON_ARCHIVE_INSTANCE_TYPE : var.ERIGON_INSTANCE_TYPE
+
+  boot_disk {
+    initialize_params {
+      image = var.INSTANCE_IMAGE
+      size  = (count.index >= var.ERIGON_VALIDATOR_COUNT + var.ERIGON_SENTRY_COUNT) ? var.ERIGON_ARCHIVE_DISK_SIZE_GB : var.ERIGON_DISK_SIZE_GB
+      type  = (count.index >= var.ERIGON_VALIDATOR_COUNT + var.ERIGON_SENTRY_COUNT) ? var.ERIGON_ARCHIVE_VOLUME_TYPE : var.ERIGON_VOLUME_TYPE
+    }
+  }
+
+  metadata = {
+    ssh-keys = "${var.USER}:${file(var.GCE_PUB_KEY_FILE)}"
+  }
+
+  network_interface {
+    network    = google_compute_network.vpc_network.name
+    subnetwork = google_compute_subnetwork.public-subnetwork.name
+
+    access_config {
+      //IP
+      nat_ip = google_compute_address.erigon_static_ip[count.index].address
+    }
+  }
+  tags = ["matic-cli"]
+  labels = {
+    name     = "polygon-matic",
+    instance = "erigon"
+
+  }
+}
+
+
+resource "google_compute_instance" "dockerized_server" {
+
+  count = (var.DOCKERIZED == "yes") ? 1 : 0
+
+  name         = "${var.VM_NAME}-docker-${count.index + 1}"
+  machine_type = var.BOR_INSTANCE_TYPE
+
+  boot_disk {
+    initialize_params {
+      image = var.INSTANCE_IMAGE
+      size  = var.BOR_DISK_SIZE_GB
+      type  = var.BOR_VOLUME_TYPE
+    }
+  }
+
+  metadata = {
+    ssh-keys = "${var.USER}:${file(var.GCE_PUB_KEY_FILE)}"
+  }
+
+  network_interface {
+    network    = google_compute_network.vpc_network.name
+    subnetwork = google_compute_subnetwork.public-subnetwork.name
+
+    access_config {
+      //IP
+      nat_ip = google_compute_address.docker_static_ip[count.index].address
+    }
+  }
+  tags = ["matic-cli"]
+  labels = {
+    name = "polygon-matic",
+    instance = "docker"
+  }
+}
+
+
+resource "google_compute_firewall" "firewall_rules" {
+  name    = format("%s-%s", var.VM_NAME, var.FW_RULE_SUFFIX)
+  network = google_compute_network.vpc_network.name
+
+  allow {
+    protocol = "tcp"
+    ports    = var.PORTS_IN
+  }
+  source_ranges = var.SG_CIDR_BLOCKS
```

Comment:
this will allow self communication between VMs, right? The `instance-0` in `matic-cli` uses `ssh` connection to the other nodes to install the network

Reply:
We have added a FW rule, it will allow the communication between all instances within a devnet.